src/types.ts

```diff
@@ -237,11 +232,37 @@ export type Currency = {
    */
   name: string;
   /**
-   * The [[FAMILIES | family]] of the cryptocurrency
+   * array of available [[Unit | units]] for the cryptocurrency
+   */
+  units: Unit[];
+};
+
+export type CryptoCurrency = BaseCurrency & {
+  /**
+   * represents the currency type.
+   * @see {@link https://github.com/LedgerHQ/ledgerjs/blob/master/packages/cryptoassets/src/types.ts|cryptoassets types} in ledgerjs for more infos
+   */
+  type: "CryptoCurrency";
+  /**
+   * the [[FAMILIES | family]] of the crypto currency
    */
   family: string;
+};
+
+export type ERC20TokenCurrency = BaseCurrency & {
   /**
-   * The array of available [[Unit | units]] for the cryptocurrency
+   * represents the currency type.
+   * @see {@link https://github.com/LedgerHQ/ledgerjs/blob/master/packages/cryptoassets/src/types.ts|cryptoassets types} in ledgerjs for more infos
    */
-  units: Unit[];
+  type: "ERC20TokenCurrency";
+  /**
+   * EVM contract address
+   */
+  contract: string;
```

Comment:
What's the use of this contract field / how would it be used / what is it's purpose?

Wouldn't there be some sort of security issue if this is provided "on the flight" by the SDK user? 🤔